RDKEMW-5996 - Addressing Possible memory leaks reported

gururaajar opened the pull request:
Reason for Change: Addressed the memory leak issues

gururaajar committed:
RDKEMW-5996 - RBUS Coverity Improvements

Reason for Change: Addressed the coverity issues

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
5294d7af1a708c52dd7e165dac7640c5
PRESENT
-->

# Coverity Issue - Unused value
Assigning value "RBUSCORE_ERROR_INSUFFICIENT_MEMORY" to "ret" here, but that stored value is overwritten before it can be used.

_Low Impact, CWE-563_
_UNUSED_VALUE_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[src/core/rbuscore.c:1946](undefined//blob/undefined/src/core/rbuscore.c#L1946)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
4d5a975c13e970e2ba338284f8e654dc
PRESENT
-->

# Coverity Issue - Unused value
Assigning value "RBUSCORE_ERROR_GENERAL" to "ret" here, but that stored value is overwritten before it can be used.

_Low Impact, CWE-563_
_UNUSED_VALUE_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[src/core/rbuscore.c:1938](undefined//blob/undefined/src/core/rbuscore.c#L1938)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
72b9bb40fd58aba8d769758386b1a72e
PRESENT
-->

# Coverity Issue - Unused value
Assigning value "NULL" to "obj" here, but that stored value is overwritten before it can be used.

_Low Impact, CWE-563_
_UNUSED_VALUE_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[src/core/rbuscore.c:3071](undefined//blob/undefined/src/core/rbuscore.c#L3071)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
6aca8633ffe52ad27c4348254564f7d2
PRESENT
-->

# Coverity Issue - String not null terminated
Passing unterminated string "new_header.topic" to "_rtdirect_send_reply_to_client", which expects a null-terminated string.

_High Impact, CWE-170_
_STRING_NULL_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[src/rtmessage/rtrouteBase.c:253](undefined//blob/undefined/src/rtmessage/rtrouteBase.c#L253)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
136c14b8dd258e140047566e1f5ec69a
PRESENT
-->

# Coverity Issue - Unchecked return value from library
Calling "remove(cacheFileName)" without checking return value. This library function may fail and return an error code.

_Medium Impact, CWE-252_
_CHECKED_RETURN_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[src/core/rbuscore.c:2615](undefined//blob/undefined/src/core/rbuscore.c#L2615)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
faf05d9fd39de4b86b58eb937fae3cfd
PRESENT
-->

# Coverity Issue - Unchecked return value from library
Calling "remove(cacheFileName)" without checking return value. This library function may fail and return an error code.

_Medium Impact, CWE-252_
_CHECKED_RETURN_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[src/core/rbuscore.c:2492](undefined//blob/undefined/src/core/rbuscore.c#L2492)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
1405baa736f38415dcdeb2e6960fc634
PRESENT
-->

# Coverity Issue - Unused value
Assigning value "RBUSCORE_ERROR_INSUFFICIENT_MEMORY" to "ret" here, but that stored value is overwritten before it can be used.

_Low Impact, CWE-563_
_UNUSED_VALUE_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[src/core/rbuscore.c:2296](undefined//blob/undefined/src/core/rbuscore.c#L2296)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
4340720b2d1c78c8b5dae52ec43c16ac
PRESENT
-->

# Coverity Issue - Overflowed constant
Expression "i--", where "i" is known to be equal to 0, underflows the type of "i--", which is type "size_t".

_Medium Impact, CWE-190_
_INTEGER_OVERFLOW_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[src/rtmessage/rtrouted.c:1498](undefined//blob/undefined/src/rtmessage/rtrouted.c#L1498)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
c735ba8c1b230c85e0a7183f7e0ac84c
PRESENT
-->

# Coverity Issue - Unchecked return value from library
Calling "setsockopt(listener->fd, 6, 1, &one, 4U)" without checking return value. This library function may fail and return an error code.

_Medium Impact, CWE-252_
_CHECKED_RETURN_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[src/rtmessage/rtrouteBase.c:93](undefined//blob/undefined/src/rtmessage/rtrouteBase.c#L93)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
e82984b90fec5e7f46fa897694519037
PRESENT
-->

# Coverity Issue - Unused value
Assigning value "RBUSCORE_ERROR_INSUFFICIENT_MEMORY" to "ret" here, but that stored value is overwritten before it can be used.

_Low Impact, CWE-563_
_UNUSED_VALUE_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[src/core/rbuscore.c:2036](undefined//blob/undefined/src/core/rbuscore.c#L2036)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
6aca8633ffe52ad27c4348254564f7d2
PRESENT
-->

# Coverity Issue - String not null terminated
Passing unterminated string "new_header.reply_topic" to "_rtdirect_send_reply_to_client", which expects a null-terminated string.

_High Impact, CWE-170_
_STRING_NULL_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[src/rtmessage/rtrouteBase.c:253](undefined//blob/undefined/src/rtmessage/rtrouteBase.c#L253)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
e5d17214f79f2e16193905f01fae172d
PRESENT
-->

# Coverity Issue - Unused value
Assigning value "RBUSCORE_ERROR_GENERAL" to "ret" here, but that stored value is overwritten before it can be used.

_Low Impact, CWE-563_
_UNUSED_VALUE_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[src/core/rbuscore.c:2028](undefined//blob/undefined/src/core/rbuscore.c#L2028)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
c735ba8c1b230c85e0a7183f7e0ac84c
PRESENT
-->

# Coverity Issue - Unchecked return value from library
Calling "setsockopt(listener->fd, 1, 2, (char *)&one, 4U)" without checking return value. This library function may fail and return an error code.

_Medium Impact, CWE-252_
_CHECKED_RETURN_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[src/rtmessage/rtrouteBase.c:95](undefined//blob/undefined/src/rtmessage/rtrouteBase.c#L95)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
32987db1ffa46fa23e2da8045ccb9207
PRESENT
-->

# Coverity Issue - Unchecked return value from library
Calling "setsockopt(con->fd, 6, 1, &i, 4U)" without checking return value. This library function may fail and return an error code.

_Medium Impact, CWE-252_
_CHECKED_RETURN_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[src/rtmessage/rtConnection.c:372](undefined//blob/undefined/src/rtmessage/rtConnection.c#L372)

rdkcmf-jenkins commented:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
c6e5754dee9cf94b9f3292f246d694bf
PRESENT
-->

# Coverity Issue - Unused value
Assigning value "RBUSCORE_ERROR_GENERAL" to "ret" here, but that stored value is overwritten before it can be used.

_Low Impact, CWE-563_
_UNUSED_VALUE_



## Issue location
This issue was discovered outside the diff for this Pull Request. You can find it at:
[src/core/rbuscore.c:2288](undefined//blob/undefined/src/core/rbuscore.c#L2288)

rdkcmf-jenkins commented on src/rtmessage/rtrouteBase.c:
```diff
@@ -524,14 +524,14 @@ rtRouteDirect_SendMessage(const rtPrivateClientInfo* pClient, uint8_t const* pIn
 
-        memset(buffer, 0,RTMSG_CLIENT_READ_BUFFER_SIZE); 
+        memset(buffer, 0, RTMSG_CLIENT_READ_BUFFER_SIZE);
         rtMessageHeader_Encode(&new_header, buffer);
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
c0dc6df2ba27e69cbfc63e7bd76cd2f9
PRESENT
-->

# Coverity Issue - String not null terminated
Passing unterminated string "new_header.topic" to "rtMessageHeader_Encode", which expects a null-terminated string.

_High Impact, CWE-170_
_STRING_NULL_

karuna2git committed:
Updated CodeQL reported issues

Updated CodeQL reported issues

karuna2git committed:
Update for the Coverity issue reported

Update for the Coverity issue reported

karuna2git committed:
Update rtrouted.c

Fixed the coverity warning `Expression "i--", where "i" is known to be equal to 0, underflows the type of "i--", which is type "size_t".`
